plugin/keyring_vault/tests/mtr/mount_point_service.inc

```diff
@@ -1,69 +1,93 @@
+  if (!$imported_vault_conf_address)
   {
-    if ($row =~ m/token[ ]*=[ ]*(.*)/)
-    {
-      $token=$1;
-    }
-    elsif ($row =~ m/vault_url[ ]*=[ ]*(.*)/)
-    {
-      $vault_url=$1;
-    }
-    elsif ($row =~ m/secret_mount_point[ ]*= [ ]*(.*)\/server[1..2]/)
-    {
-      $secret_mount_point=$1;
-    }
-    elsif ($row =~ m/secret_mount_point[ ]*= [ ]*(.*)/)
-    {
-      $secret_mount_point=$1;
-    }
-    elsif ($row =~ m/vault_ca[ ]*= [ ]*(.*)/)
-    {
-      $vault_ca=$1;
-    }
+    $imported_vault_conf_address = $ENV{'MTR_VAULT_ADDRESS'} or die "Please export MTR_VAULT_ADDRESS variable";
   }
-  close(CONF_FILE);
-  if ($token eq "" || $vault_url eq "" || $secret_mount_point eq "")
+
+  my $imported_vault_admin_token = $ENV{'EXPORTED_VAULT_ADMIN_TOKEN'};
+  if (!$imported_vault_admin_token)
+  {
+    $imported_vault_admin_token = $ENV{'MTR_VAULT_ADMIN_TOKEN'} or die("Please export MTR_VAULT_ADMIN_TOKEN variable");
+  }
+
+  my $imported_vault_conf_ca = $ENV{'EXPORTED_VAULT_CONF_CA'};
+  if (!$imported_vault_conf_ca)
+  {
+    $imported_vault_conf_ca = $ENV{'MTR_VAULT_CA'};
+  }
+
+  my $imported_vault_conf_mount_point_uuid = $ENV{'EXPORTED_VAULT_CONF_MOUNT_POINT_UUID'} or die("EXPORTED_VAULT_CONF_MOUNT_POINT_UUID not set\n");
+  my $imported_vault_conf_mount_point_suffix = $ENV{'EXPORTED_VAULT_CONF_MOUNT_POINT_SUFFIX'};
+  my $imported_vault_conf_mount_point_directory = $ENV{EXPORTED_VAULT_CONF_MOUNT_POINT_DIRECTORY};
+  my $imported_vault_server_mount_point_version = $ENV{EXPORTED_VAULT_SERVER_MOUNT_POINT_VERSION};
+  my $imported_mount_point_service_op = $ENV{'EXPORTED_MOUNT_POINT_SERVICE_OP'};
+
+  my $vault_mount = $imported_vault_conf_mount_point_uuid;
+  if (!$imported_vault_conf_mount_point_directory)
   {
-    die("Could not read vault credentials from configuration file.\n");
+    if ($imported_vault_conf_mount_point_suffix)
+    {
+      $vault_mount .= $imported_vault_conf_mount_point_suffix;
+    }
   }
 
-  my $vault_ca_cert_opt= "";
-  if ($vault_ca)
+  my $vault_ca_cert_opt = "";
+  if ($imported_vault_conf_ca)
   {
-    $vault_ca_cert_opt= "--cacert $vault_ca";
+    $vault_ca_cert_opt = "--cacert $imported_vault_conf_ca";
   }
 
-  if ($mount_point_service_op eq 'CREATE')
+  if ($imported_mount_point_service_op eq 'CREATE')
   {
-    system(qq#curl --http1.1 -L -H "X-Vault-Token: $token" $vault_ca_cert_opt --data '{"type":"generic"}' --request POST $vault_url/v1/sys/mounts/$secret_mount_point#);
+    my $postdata= "";
+    if ($imported_vault_server_mount_point_version eq "2")
+    {
+      $postdata= qq#{"type":"kv","options":{"version":"2"}}#;
+    } else {
+      $postdata= qq#{"type":"kv"}#;
+    }
+    system(qq#curl --http1.1 --insecure -L -H "X-Vault-Token: $imported_vault_admin_token" $vault_ca_cert_opt --data '$postdata' --request POST $imported_vault_conf_address/v1/sys/mounts/$vault_mount#) == 0
+      || die "Cannot create mount point";
```

Comment:
I'd suggest using "or die ..." here instead. The "or" and "||" have different precedence in Perl and only "or" is usually used for flow control to avoid unexpected behavior.

Reply:
fixed